data_handler.py

```diff
@@ -140,12 +151,46 @@ def main(self, datalog, cpm, cpm_err, this_start, this_end, counts):
         self.manager.data_log(datalog, cpm, cpm_err)
 
         if self.manager.test:
-            self.test_send(cpm, cpm_err)
+            self.no_network_send(cpm, cpm_err)
         elif not self.manager.config:
             self.no_config_send(cpm, cpm_err)
         elif not self.manager.publickey:
             self.no_publickey_send(cpm, cpm_err)
-        elif not self.manager.network_up:
-            self.no_network_send(cpm, cpm_err)
         else:
-            self.regular_send(this_end, cpm, cpm_err)
+            try:
+                self.regular_send(this_end, cpm, cpm_err)
+            except socket.gaierror as e:
+                if e[0] == socket.EAI_AGAIN:
+                    # TCP and UDP
+                    # network is down, but NetworkStatus didn't notice yet
+                    # (resolving DNS like dosenet.dhcp.lbl.gov)
+                    self.vprint(
+                        1, 'Failed to send packet! Address resolution error')
+                    self.no_network_send(cpm, cpm_error)

```

Comment:
Also, you have `self.no_network_send` in every condition of the `if` statement, so you could just put it once after the `if` block for all of them to execute. Or before the `if` block.